cockroachcloud/client-certs-dedicated.md

```diff
@@ -0,0 +1,332 @@
+- [Create certificates with `cockroach cert`](../{{site.versions["stable"]}}/cockroach-cert.html#synopsis)
+- [Create Security Certificates using OpenSSL](../{{site.versions["stable"]}}/create-security-certificates-openssl.html)
+
+### Initialize your Vault workstation
+
+1. [Install Vault](https://learn.hashicorp.com/tutorials/vault/getting-started-install) on your workstation. Consider that your workstation must be secure in order for the PKI hierarchy you are establishing to be secure. Consider using a dedicated secure jumpbox, as described in [PKI Strategy](../{{site.versions["stable"]}}/manage-certs-vault.html#pki-strategy).
+
+1. Obtain the required parameters to target and authenticate to Vault.
+    
+    1. Option 1: If using a development Vault server (*suitable only for tutorial/testing purposes*), start the Vault development server and obtain your admin token locally on your CA admin jumpbox by running `vault server --dev`.
+
+    1. Option 2: If using HashiCorp Cloud Platform (HCP):
+
+        1. Go to the [HCP console](https://portal.cloud.hashicorp.com), choose Vault from the **Services** menu and then select your cluster.
+
+        1. Find the **Public Cluster URL** for your Vault cluster. This will be set as the `VAULT_ADDR` environment variable, in the following step.
+
+        1. Generate an admin token by clicking **Generate token**. This will be set as the `VAULT_TOKEN` environment variable, in the following step.
+
+    1. Option 3: If using a Vault deployment internal to your organization, contact your Vault administrator for a token and the appropriate endpoint.
+
+1. Initialize your shell for Vault.
+
+    {% include_cached copy-clipboard.html %}
+    ~~~shell
+    export VAULT_ADDR= # your Vault cluster's Public URL
+    export VAULT_NAMESPACE="admin"
+    ~~~
+
+1. Authenticate with your admin token 
+
+    {% include_cached copy-clipboard.html %}
+    ~~~shell
+    vault login
+    ~~~
+
+### Create the certificate authority (CA) certificate
+
+This CA certificate will be used to [configure your cluster's Trust Store](#upload-a-certificate-authority-ca-certificate-for-a-cockroachdb-dedicated-cluster). Any client certificate signed by the CA identified by this certificate will be trusted, and be able to authenticate to your cluster.
+
+1. Create a PKI secrets engine to serve as your client CA.
+
+    {% include_cached copy-clipboard.html %}
+    ~~~shell
+    vault secrets enable -path=cockroach_client_ca pki
+    ~~~
+
+    ~~~txt
+    Success! Enabled the pki secrets engine at: cockroach_client_ca/
+    ~~~
+    
+1. Generate a root credential pair for the CA. Certificates created with this CA/secrets engine will be signed with the private key generated here and held within Vault; this key cannot be exported, safeguarding it from being leaked and used to issue fraudulent certificates. The CA public certificate is downloaded in the resulting JSON payload.
```

Comment:
I prefer mine, is this a strong opinion?

Reply:
At the very least, I think the guarantee needs to be weakened. We can't guarantee that the CA won't be leaked just because they use Vault. That's what "help" is doing in my suggestion. Also, "safeguarding" seems weird and jargony as a verb.

Reply:
"Safeguarding" is not jargon, it's regular english and not particularly obscure. Our readers can look an unfamiliar word up in the dictionary if they need to. I think the definitions offered by oxford or merriam webster are perfectly appropriate to what I mean. They do not imply a guarantee, just a measure intended for protection or precaution.

It's Vault's guarantee, not ours, that the key can't be exported:
https://developer.hashicorp.com/vault/docs/secrets/pki/setup
"The private key is safely stored internally in Vault."

Reply:
SGTM